source/code/omi_preexec/omi-mysql.cpp

```diff
@@ -82,113 +83,19 @@ static int MySQL_ConfigurationParser(uid_t uid, gid_t gid, const std::wstring& c
         return 0;
     }
 
-    SCX_LOGHYSTERICAL(hLog, StrAppend(L"Parsing MySQL configuration file: ", cnfFile));
+    SCX_LOGHYSTERICAL(hLog, StrAppend(L"Loading MySQL configuration file: ", cnfFile));
 
-    std::vector<std::wstring> lines;
-    try
+    MySQL::MySQL_ConfigurationScanner configScanner;
+    if (!configScanner.Scan(cnfFile, auth))
     {
-        SCXStream::NLFs nlfs;
-        SCXFile::ReadAllLines(cnfFile, lines, nlfs );
-    }
-    catch (const SCXException& e)
-    {
-        SCX_LOGERROR(hLog, StrAppend(L"Error reading configuration file: ", e.What()));
+        SCX_LOGERROR(hLog, StrAppend(L"Error reading ", cnfFile));
```

Comment:
This is a poor error. If your class can throw, then you're responsible for having meaningful exceptions, and you should at least say what the exception is (like I used to do in the old code). But this assumes that your class raises meaningful exceptions in case of errors.

Seeing an error like: "Error reading <filename>" is not helpful to a customer.